tests/python/unittest/test_runtime_micro.py

```diff
@@ -0,0 +1,183 @@
+    a = tvm.nd.array(np.random.uniform(size=shape).astype(dtype), ctx)
+    b = tvm.nd.array(np.random.uniform(size=shape).astype(dtype), ctx)
+    c = tvm.nd.array(np.zeros(shape, dtype=dtype), ctx)
+    micro_func(a, b, c)
+
+    tvm.testing.assert_allclose(
+        c.asnumpy(), a.asnumpy() + b.asnumpy())
+
+
+def test_workspace_add():
+    """Test a program which uses a workspace."""
+    shape = (1024,)
+    dtype = "float32"
+
+    # Construct TVM expression.
+    tvm_shape = tvm.convert(shape)
+    A = tvm.placeholder(tvm_shape, name="A", dtype=dtype)
+    B = tvm.placeholder(tvm_shape, name="B", dtype=dtype)
+    B = tvm.compute(A.shape, lambda *i: A(*i) + 1, name="B")
+    C = tvm.compute(A.shape, lambda *i: B(*i) + 1, name="C")
+    s = tvm.create_schedule(C.op)
+
+    func_name = "fadd_two_workspace"
+    host_mod = tvm.build(s, [A, C], target="c", name=func_name)
+
+    micro.init(DEVICE_TYPE)
+    micro_mod = micro.from_host_mod(host_mod, DEVICE_TYPE)
+    micro_func = micro_mod[func_name]
+    ctx = tvm.micro_dev(0)
+    a = tvm.nd.array(np.random.uniform(size=shape).astype(dtype), ctx)
+    c = tvm.nd.array(np.zeros(shape, dtype=dtype), ctx)
+    micro_func(a, c)
+
+    tvm.testing.assert_allclose(
+        c.asnumpy(), a.asnumpy() + 2.0)
+
+
+def test_graph_runtime():
+    """Test a program which uses the graph runtime."""
+    shape = (1024,)
+    dtype = "float32"
+
+    # Construct Relay program.
+    x = relay.var("x", relay.TensorType(shape=shape, dtype=dtype))
+    xx = relay.multiply(x, x)
+    z = relay.add(xx, relay.const(1.0))
+    func = relay.Function([x], z)
+
+    micro.init(DEVICE_TYPE)
+    mod, params = relay_micro_build(func)
+
+    mod.set_input(**params)
+    x_in = np.random.uniform(size=shape[0]).astype(dtype)
+    mod.run(x=x_in)
+    result = mod.get_output(0).asnumpy()
+
+    tvm.testing.assert_allclose(
+        result, x_in * x_in + 1.0)
+
+
+def test_resnet_random():
+    """Test ResNet18 inference with random weights and inputs."""
+    resnet_func, params = resnet.get_workload(num_classes=10,
+                                              num_layers=18,
+                                              image_shape=(3, 32, 32))
+    # Remove the final softmax layer, because uTVM does not currently support
+    # it.
+    resnet_func_no_sm = relay.Function(resnet_func.params,
+                                       resnet_func.body.args[0],
+                                       resnet_func.ret_type)
+    micro.init(DEVICE_TYPE)
+    # TODO(weberlo): Use `resnet_func` once we have libc support.
+    mod, params = relay_micro_build(resnet_func_no_sm, params=params)
+    mod.set_input(**params)
+    # Generate random input.
+    data = np.random.uniform(size=mod.get_input(0).shape)
+    mod.run(data=data)
+    result = mod.get_output(0).asnumpy()
+    # We gave a random input, so all we want is a result with some nonzero
+    # entries.
+    assert result.sum() != 0.0
+
+
+def test_resnet_pretrained():
```

Comment:
Let us remove the end to end test for now

Reply:
please update per comment

Reply:
@tqchen Is adding the `@nottest` decorator sufficient?  If not, I'd still like to have this code somewhere, even if on a branch on a private repo.